### PR TITLE
Added type and rarity to popup and Pokemondb.net to pokedex links

### PR DESCRIPTION
**Pull request by chaze117**
## Description
Added the code written by  [UzEE](https://github.com/UzEE) in [#2109](https://github.com/AHAAAAAAA/PokemonGo-Map/pull/2109) and I added pokemon rarity based on [this](http://i.imgur.com/Qmx2vF2.jpg) image.
## Motivation and Context
I really liked what UzEE did and I added a little extension. All thank goes to [UzEE](https://github.com/UzEE).
## How Has This Been Tested?
I run my server with this extension  by a couple days and it works fine on PC and Android too.

## Screenshots (if appropriate):
pgomap
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: static/dist/js/map.js [hunk 1]
```diff
@@ -355,11 +355,28 @@ function initSidebar() {
 function pad(number) {
   return number <= 99 ? ("0" + number).slice(-2) : number;
 }
+var typesMap = {'bug':'#a8b820','dark':'#705848','dragon':'#7038f8','electric':'#f8d030','fairy':'#e898e8','fighting':'#c03028','fire':'#f08030','flying':'#a890f0','ghost':'#705898','grass':'#78c850','ground':'#e0c068','ice':'#98d8d8','normal':'#8a8a59','poison':'#a040a0','psychic':'#f85888','rock':'#b8a038','steel':'#b8b8d0','water':'#6890f0'};
```
Comment (chaze117) [inline static/dist/js/map.js]: This code made by [UzEE](https://github.com/UzEE)

**Comment by joveice** [comment]
+1

**Comment by invisiblek** [comment]
Hmm, make sure you're changing the static/map.js and static/js/stats.js

You should never directly edit static/dist/*, those are generated (minified) versions created by grunt.

This is an awesome feature though!

**Comment by chaze117** [comment]
Uhhhm okay, but some develop braches ago I edited static/map.js and the changes didn't worked, thats why I'm edited the static/dist files

**Comment by invisiblek** [comment]
You need to edit static/map.js and then run grunt to generate static/dist/js/map.js

**Comment by chaze117** [comment]
Then I will close this and do it the rigth way . Thank you

**Comment by andrewda** [comment]
Also, could you get the rarity using the percentage of that Pokemon found in the area? There are already a ton of PRs to do something like that. The chart is very inaccurate because it all depends on location.

**Comment by andrewda** [comment]
For example, the chart shows Tauros as Epic and Pinsir as Rare, but I've caught at least 20 of each in the past couple days. In my area, they're two of the most common.

**Comment by chaze117** [comment]
Yeah, but the method that calculates the rarity in a specific area will be good, unless you place the marker elsewhere. And the problem comes out here, new location new common pokemons but in the db still there the old location's pokemons with the old statistic and the whole thing messed up.

By the way I just started this whole thing not too good at python and js, I mostly use C# and php but I try to look after what I can do.

**Comment by tim95030** [comment]
I have added rarities in this pull request #1899 which is already marked for the next milestone, so I wouldn't spend time re-implementing this fyi. If you prefer those rarities to the ones I gathered you are welcome to submit a change after mine is merged, but I know the hope is to involve actual statistics of local data if others agree as with the stats platform starting we have a better place to dynamically draw rarity information.

**Pull request closed without merging**